### PR TITLE
DCS-502 Create Typescript interface(s) that describe the structure of licences.licence database objects.

### PR DESCRIPTION
**Pull request by bell-pa**

### CODE REVIEW

File: server/services/licenceService.ts [hunk 1]
```diff
@@ -22,39 +24,54 @@ const {
...
+        return null
+      }
 
-      const licence = getIn(rawLicence, ['licence'])
+      const { licence, stage } = rawLicence
       if (!licence) {
         return null
       }
-      const formattedLicence = formatObjectForView(licence)
-      const approvedVersionDetails = rawVersionDetails ? formatObjectForView(rawVersionDetails) : {}
```
Comment (bell-pa) [inline server/services/licenceService.ts:50]: I can't see how {formatObjectForView} could change {licence} or {rawVersionDetails} so I've removed both invocations.  All the tests are passing, I've tried checking the code for the field names used within that function and haven't found any matches.  

Is this risky???

File: server/services/licenceService.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
-const moment = require('moment')
```
Comment (bell-pa) [inline server/services/licenceService.ts:1]: This class / file is too big, but I'll leave dismembering it for another time.

File: server/data/licenceClient.ts [hunk 3]
```diff
@@ -119,8 +151,9 @@ module.exports = {
     }
 
     return db.query(query)
-  },
+  }
 
+  // eslint-disable-next-line class-methods-use-this
```
Comment (andrewrlee) [inline server/data/licenceClient.ts:156]: would be tempted to move to inject `db` in the future, then we can use normal mocking in the db tests and not have these warnings to suppress. It's a bit random we actually require it directly

File: server/data/licenceClientTypes.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { Licence } from './licenceTypes'
+
+export interface Case {
+  licence: Licence
+  booking_id: number
```
Comment (andrewrlee) [inline server/data/licenceClientTypes.ts:5]: If we get more TS coverage we could map these names using a column alias to camelCase

File: server/data/licenceClient.ts [hunk 5]
```diff
@@ -41,12 +60,13 @@ module.exports = {
       return rows[0]
     }
 
-    return {}
-  },
+    return undefined
```
Comment (andrewrlee) [inline server/data/licenceClient.ts:63]: I can only see this being used in one place, is that right?

**Pull request merged**